package/src/utils/utils.ts

```diff
@@ -168,25 +160,28 @@ const getWatchers = <
   channel: Channel<StreamChatGenerics>,
 ) => {
   const watchers = channel.state.watchers;
-  return Object.values(watchers).length ? [...Object.values(watchers)] : [];
+  return watchers ? Object.values(watchers) : [];
 };
 
 const getMembersAndWatchers = <
   StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
 >(
   channel: Channel<StreamChatGenerics>,
 ) => {
-  const users = [...getMembers(channel), ...getWatchers(channel)];
+  const members = getMembers(channel);
+  const watchers = getWatchers(channel);
+  const users = [...members, ...watchers];
 
-  return Object.values(
-    users.reduce((acc, cur) => {
-      if (!acc[cur.id]) {
-        acc[cur.id] = cur;
-      }
+  // make sure we don't list users twice
+  const uniqueUsers = {} as Record<string, UserResponse<StreamChatGenerics>>;
+
+  users.forEach((user) => {
+    if (user && !uniqueUsers[user.id]) {
+      uniqueUsers[user.id] = user;
+    }
+  });
```

Comment:
Unrelated, but I think you can speed up this function and save one full object iteration (`Object.values()`) if you rewrite it like this:

```ts
const seenUsers = new Set<string>();
const uniqueUsers: User[] = [];
for (const user of users) {
  if (user && !seenUsers.has(user.id)) {
    uniqueUsers.push(user);
    seenUsers.add(user.id);
  }
}
return uniqueUsers;
```